File: library/src/main/java/com/uploadcare/android/library/urls/UrlUtils.java
package com.uploadcare.android.library.urls;

import android.net.Uri;

import java.net.URI;
import java.net.URISyntaxException;

public class UrlUtils {

    public static URI trustedBuild(Uri.Builder builder) {
        try {
            return new URI(builder.build().toString());
        } catch (URISyntaxException e) {
            throw new IllegalArgumentException(e);
        }
    }

}

File: library/src/main/java/com/uploadcare/android/library/urls/UrlUtils.kt
package com.uploadcare.android.library.urls

import android.net.Uri
import java.net.URI
import java.net.URISyntaxException

class UrlUtils {
    companion object {

        fun trustedBuild(builder: Uri.Builder): URI {
            try {
                return URI(builder.build().toString())
            } catch (e: URISyntaxException) {
                throw IllegalArgumentException(e)
            }
        }
    }
}